### DISCUSSION TITLE
ThingsDB PHP driver

### DISCUSSION THREAD

**Discussion by stefanak-michal** [thingsdb/ThingsDB#370] [Show and tell]
Hi.

Let me share with you PHP driver for communication with ThingsDB over socket I've made. Please enjoy.

https://github.com/stefanak-michal/thingsdb-php

**Comment by joente** [comment]
If you're okay, I'd like to add your PHP connector to the ThingsDB documentation (and book).

**Reply by stefanak-michal** [reply]
Of course, no problem. I'm also planning to make js driver, but I'm waiting for some news if websocket will be added.

**Reply by joente** [reply]
I'm working on it.

**Reply by stefanak-michal** [reply]
That's great. I didn't expect that fast and positive response.

**Comment by joente** [comment]
Did you consider making the library compatible with PHP 8.1? (This seems to be the default version in Ubuntu 22.04 LTS). Currently, the use of `readonly class` is preventing 8.1 support, or is there more?

**Reply by stefanak-michal** [reply]
I like to keep up and I try to follow with https://www.php.net/supported-versions.php

8.1 is receiving only security support until the end of this year. Anyway it shouldn't be problem to update from 8.1 to 8,2.

It is not only readonly class. I use phpunit 11 which requires php 8.2.

**Comment by stefanak-michal** [comment]
I wrote article about it here https://dev.to/stefanak-michal/thingsdb-and-php-3c0k
